### ISSUE TITLE
centos7 support? 

### ISSUE THREAD

**Issue by jiamo** [utoni/nDPId#20]
with cmake installed, nDPId seem can't support centos 7 default build system while libndpi is ok. 
have such error 
```
/home/jiamo/diting_ndpid/dependencies/nDPIsrvd.h: In function 'nDPIsrvd_get_next_token':
/home/jiamo/diting_ndpid/dependencies/nDPIsrvd.h:920:5: error: 'for' loop initial declarations are only allowed in C99 mode
     for (int i = *next_index + 1; i < sock->jsmn.tokens_found; ++i)
```
But if I cmake like `cmake  -DBUILD_NDPI=off  -DCMAKE_C_FLAGS="-std=c99" ..` got something like
```
/home/jiamo/diting_ndpid/examples/c-captured/c-captured.c:663:34: error: 'optarg' undeclared (first use in this function)
                 pidfile = strdup(optarg);
                                  ^
```
I know gcc in centos 7  is too low.  Is there some good method can make centos7 work ?

**Comment by utoni** [comment]
Ok, I'll add CentOS to the CI, see what errors occur and try to fix them. Thanks for reporting!

**Comment by utoni** [comment]
Adding CentOs7 to Github Actions is not as easy as I initially thought. Maybe because it is kinda old and support gets dropped anyway next year.

Meanwhile, I've might have found a solution to your issue;
Can you please try to compile `nDPId` from the branch `tmp` and give me feedback if it works for you. Thanks!

**Comment by utoni** [comment]
@jiamo kindly pinging..

**Comment by jiamo** [comment]
I got such error 
```
CMakeFiles/nDPId.dir/nDPId.c.o: In function `ndpi_debug_printf':
/home/jiamo/nDPId/nDPId.c:1215: undefined reference to `ndpi_get_user_data'
CMakeFiles/nDPId.dir/nDPId.c.o: In function `init_workflow':
/home/jiamo/nDPId/nDPId.c:1330: undefined reference to `ndpi_set_user_data'
collect2: error: ld returned 1 exit status
make[2]: *** [nDPId] Error 1
make[1]: *** [CMakeFiles/nDPId.dir/all] Error 2
```

**Comment by jiamo** [comment]
I am using the latest ndpi. after `cmake -DBUILD_NDPI=on ..` It works!

**Comment by utoni** [comment]
Thanks for testing! I'll finish my work on branch `tmp` and will then merge it to `main` and close this issue. :)

**Comment by utoni** [comment]
Fixed since d0c070a800c7577e6d437812f0d3bd976ddc475e.
